### PR TITLE
UI: Deposit Details step

### PR DESCRIPTION
**Pull request by kpyszkowski**
Ref: #5  
~Depends on: #12~ 

This pull request introduces redesigned Deposit Details minting flow step redesign.
localhost_3000_tBTC_mint (9)

New components: _#TODO: Add screenshots_ 
- `BridgeProcessIndicator` component
- `BridgeProcessResource` component
- `TransactionHistory` component
- `BridgeProcessCircularLoader` component

### CODE REVIEW

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx:26]: The links for those two contracts are wrong. Please see how we build those links based on the connected chain here:

https://github.com/threshold-network/bitcoin-on-base/blob/6dc91ef4fa6bd9794f26ef5dcecf06c39b6ebf66/src/components/tBTC/Links.tsx#L14-L43

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:25]: Let's remove `$nbsp;` here and replace it with `{" "}` (or with the previous implementation). We don't want non-breaking space here.

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:35]: Non-blocking:

While we're at it can we also add `DepositDetails` prefix to step components? Like `DepositDetailsStep1`, `DepositDetailsStep2` etc.?

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments to look at before the merge

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:78]: Hmm, loks like loader progress doesn't work properly. For Step 3 it should be at 50%, but it's at 25:

image

I've also had 25 at step1, when it should be 0.

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:51]: I've also noticed that bitcoin confirmations are not updated when they happen on the chain 🤔 I'm not sure what's the issue yet though, so for now I just point that out. I had to refresh the deposit details page to display me the step 2.

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx:40]: I've actually mentioned this issue in another PR too: https://github.com/threshold-network/bitcoin-on-base/pull/37#discussion_r1511040829. Please take a look at it as we can combine the fix for both.

Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx:40]: @kpyszkowski I've actually mentioned this issue in another PR too: https://github.com/threshold-network/bitcoin-on-base/pull/37#discussion_r1511040829. Please take a look at it as we can combine the fix for both.

File: src/components/tBTC/BridgeProcessIndicator.tsx
Comment (kpyszkowski) [reply src/components/tBTC/BridgeProcessIndicator.tsx:11]: I forgot about this TODO tbh.
However I have introduce such icon components in #44 so let's complete this TODO in separate PR addressing #36

File: src/types/tbtc.ts
Comment (kpyszkowski) [reply src/types/tbtc.ts]: Ref commit: b5f6a7995b4418789a9becb997c465600d7ff247

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx]: Ref commit: 2f6222df2ea6d7c491f825e057c47445580eb3f2

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx]: Ref commit: d2cde08dc732ebbd2122ba5b5689bd826e6a55fb

File: src/pages/tBTC/Bridge/components/TransactionHistory.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/TransactionHistory.tsx]: Ref commit: d9e455998dcaeff91b743666f10312886d622cb1

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx:32]: That's how layout is expected to work - to separate main content from aside areas for separation of concerns. To accomplish it was mandatory to expose required data to be accessible outside the main content. 
The data required to render `KnowledgeBaseLinks` are `mintingRequestedTxHash` and `mintingFinalizedTxHash`. As you pointed out, it can be gathered by calling the hook that was previously called in `DepositDetails`, That way we don't have to keep it in global store. 
It it okay as is currently. I wouldn't like to change anything in this area.

File: src/pages/tBTC/Bridge/index.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/index.tsx]: Ref commit: edfe1c910efc2d713452090e1b8ca727d6ff6003

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx]: Ref commit: c599320c910f93b2a26f11341f79aa647a3d1f60

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx]: Ref commit: b4762d42af63550fa0e59c31bf3bc7ef52ec7844

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx]: Ref commit: be7d18d34a6580c841ff9af8f54e5d693b3402f8

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:90]: I have accidentally left hard coded value, the prop value wasn't passed to the function 😆 Should be fine now
Ref commit: 55f368196007890c1c923524410289ae6c72fae8

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx:57]: Hmm, that's true. Ain't got a clue why. I didn't change anything in this area. To prevent blocking this PR let's investigate in separate PR okay? The UI itself works fine.

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some more comments

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx:1]: Ref commit: f6ba0ea7

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some more comments to look at

File: src/pages/tBTC/Bridge/components/BridgeProcessResources.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessResources.tsx]: Ref commit: cd34f388

File: src/pages/tBTC/Bridge/components/BridgeProcessResources.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessResources.tsx]: Ref commit: cd34f388

File: src/pages/tBTC/Bridge/components/BridgeProcessCircularLoader.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessCircularLoader.tsx]: Ref commit: 22f33bdf

File: src/pages/tBTC/Bridge/components/BridgeProcessCircularLoader.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessCircularLoader.tsx]: Ref commit: 22f33bdf

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx]: Ref commit: 1bf4416c08c4d1d446c308f39e3cd08d472758f0

File: src/components/tBTC/BridgeProcessIndicator.tsx
Comment (kpyszkowski) [reply src/components/tBTC/BridgeProcessIndicator.tsx:6]: Ref commit: d8442e63

File: src/pages/tBTC/Bridge/components/BridgeProcessResource.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessResource.tsx:4]: Ref commit: 56d37d29

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx]: Done

File: src/pages/tBTC/Bridge/components/BridgeProcessStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessStep.tsx]: Ref commit: 1e6d99fe

File: src/pages/tBTC/Bridge/components/BridgeProcessStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/BridgeProcessStep.tsx]: Ref commit: 1e6d99fe

File: src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/Minting/KnowledgebaseLinks.tsx:1]: Ref commit: f6ba0ea7c

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx]: Already removed

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx:37]: Ref commit: 9375d4a8

File: src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/components/DepositDetailsStep.tsx]: Ref commit: 538193a97c262ab11862acf8308ab796b50a832c

**Review by michalsmiarowski — APPROVED** [review]
LGTM 

Some of the bug will be fixed in a separate PR: https://github.com/threshold-network/bitcoin-on-base/pull/45

**Pull request merged**
